translator/util/ec2util/ec2util.go

```diff
@@ -67,46 +71,51 @@ func initEC2UtilSingleton() (newInstance *ec2Util) {
 		log.Println("W! [EC2] Sleep until network is up")
 		time.Sleep(1 * time.Second)
 	}
+
 	if !networkUp {
 		log.Println("E! [EC2] No available network interface")
 	}
 
-	ses, err := session.NewSession()
+	err := newInstance.deriveEC2MetadataFromIMDS()
+
 	if err != nil {
-		log.Println("E! [EC2] getting new session info: ", err)
-		return
+		log.Println("E! [EC2] Cannot get EC2 Metadata from IMDS:", err)
 	}
-	md := ec2metadata.New(ses)
 
-	if !md.Available() {
-		log.Println("E! ec2metadata is not available")
-		return
-	}
+	return
+}
 
-	if info, err := md.GetMetadata("instance-id"); err == nil {
-		newInstance.InstanceID = info
-	} else {
-		log.Println("E! getting instance-id from EC2 metadata fail: ", err)
+func (e *ec2Util) deriveEC2MetadataFromIMDS() error {
+	ses, err := session.NewSession()
+
+	if err != nil {
+		return err
 	}
 
-	if info, err := md.GetMetadata("hostname"); err == nil {
-		newInstance.Hostname = info
-	} else {
-		log.Println("E! getting hostname from EC2 metadata fail: ", err)
+	md := ec2metadata.New(ses)
+
+	if !md.Available() {
+		return errors.New("EC2 metadata is not available.")
 	}
 
-	if info, err := md.GetMetadata("local-ipv4"); err == nil {
-		newInstance.PrivateIP = info
+	// Only need the API to scrape HostName
```

Comment:
I don't see a lot of value in this comment. It is clear from the code that you are getting the `hostname` metadata value.

Reply:
Agree. The code is straightforward enough. Therefore, I will remove it in the newest commit.